effects/nixops/run-nixops.nix

```diff
@@ -148,7 +166,7 @@ mkEffect (
   # To quote the NixOps help:
   #   check the state of the machines in the network (note that this might alter
   #   the internal nixops state to consolidate with the real state of the resource)
-  effectCheckScript = ''
+  effectCheckScript = lib.optionalString (action != "dry-run") ''
```

Comment:
It's probably better not to upload the state files either.

Reply:
Hmm, so those put and get state file functions use curl, I think they caused this false negative:

![image](https://user-images.githubusercontent.com/3874017/112991871-a0ae3e00-9135-11eb-89f3-975f46c129a1.png)

If we can disable it for dry-run I'll add that. 
This only happened once of more runs, so it's probably some race.
However I don't want to push false negatives on my colleagues, the point is sort of to make modifying nix code more accessible.

Reply:
I'm testing out how often it occurs with this:

```
for i in {0..50}; do git commit --allow-empty -m "effect false negative test "$i; git push; done
```

Maybe it's useful for you as well.

Reply:
I tried it out for 50, no more failures. I'm not sure how much trouble it is to fix this, but if the probability is less than 1/50 I'm okay with merging into master for supercede (previously known as riskbook).